### PR TITLE
fix crossentropy loss

### PR DESCRIPTION
**Pull request by XiaodongWuFetch**
1. Fix Average backprop
2. Fix incorrect logits input assumption

### CODE REVIEW

File: libs/ml/include/ml/ops/loss_functions/cross_entropy_loss.hpp [hunk 1]
```diff
@@ -72,53 +72,33 @@ class CrossEntropyLoss : public Ops<T>
...
-        ++r_it;
+        *r_it = -*b_it / *a_it;
       }
+      ++a_it;
+      ++b_it;
+      ++r_it;
     }
-
-    return {ret, ret};
+    return {ret / batch_size, ret / batch_size};
```
Comment (MissingNO57) [inline libs/ml/include/ml/ops/loss_functions/cross_entropy_loss.hpp]: Please just remove that division from incorrect gradient

**Review by MissingNO57 — CHANGES_REQUESTED** [review]

**Review by MissingNO57 — DISMISSED** [review]
LGTM

**Review by MissingNO57 — APPROVED** [review]
LGTM
